### PR TITLE
bulk-cdk-core-extract: fix catalog schema bug

### PR DESCRIPTION
**Pull request by postamar**
## What
This PR fixes https://github.com/airbytehq/airbyte-internal-issues/issues/9026 in the Bulk CDK. This PR also re-exports the imported BOMs, which is needed by connectors which use the Bulk CDK.

## User Impact
None

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/bulk/core/extract/src/test/kotlin/io/airbyte/cdk/read/StateManagerStreamStatesTest.kt [hunk 1]
```diff
@@ -185,10 +185,6 @@ class StateManagerStreamStatesTest {
         val eventsStream: Stream = stateManager.feeds.mapNotNull { it as? Stream }.first()
         Assertions.assertEquals("EVENTS", eventsStream.name)
         Assertions.assertEquals(listOf("MSG", "ID", "TS"), eventsStream.fields.map { it.id })
-        Assertions.assertEquals(
-            listOf(listOf("ID")),
-            eventsStream.primaryKeyCandidates.map { col -> col.map { it.id } },
-        )
```
Comment (evantahler) [inline airbyte-cdk/bulk/core/extract/src/test/kotlin/io/airbyte/cdk/read/StateManagerStreamStatesTest.kt:191]: Why was this test removed? It is correct for a single PK `source_defined_primary_key=[[ID]]`

File: airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/JdbcMetadataQuerier.kt [hunk 2]
```diff
@@ -299,13 +299,10 @@ class JdbcMetadataQuerier(
...
-            .groupBy { it.name }
-            .values
-            .map { rowsByPK: List<PrimaryKeyRow> ->
-                rowsByPK.sortedBy { it.ordinal }.map { it.columnName }
-            }
-            .also { memoizedPrimaryKeys[table] = it }
+        val rows: List<PrimaryKeyRow> = results.groupBy { it.name }.values.firstOrNull() ?: listOf()
+        val pk: List<List<String>> = rows.sortedBy { it.ordinal }.map { listOf(it.columnName) }
+        memoizedPrimaryKeys[table] = pk
+        return pk
```
Comment (evantahler) [inline airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/JdbcMetadataQuerier.kt:305]: I think this is the main change

File: airbyte-cdk/bulk/core/extract/src/test/kotlin/io/airbyte/cdk/read/StateManagerStreamStatesTest.kt (hunk 1 above)
Comment (postamar) [reply airbyte-cdk/bulk/core/extract/src/test/kotlin/io/airbyte/cdk/read/StateManagerStreamStatesTest.kt:191]: We no longer have the `primaryKeyCandidates` field in the `Stream`, as `configuredPrimaryKey` makes it redundant

File: airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/JdbcMetadataQuerier.kt (hunk 2 above)
Comment (postamar) [reply airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/JdbcMetadataQuerier.kt:305]: Yes

**Review by theyueli — APPROVED** [review]
thanks for the fix!

**Pull request merged**